third_party/bigframes_vendored/pandas/core/computation/eval.py

```diff
@@ -0,0 +1,401 @@
+# Contains code from https://github.com/pandas-dev/pandas/blob/main/pandas/core/computation/eval.py
+"""
+Top level ``eval`` module.
+"""
+from __future__ import annotations
+
+import tokenize
+from typing import TYPE_CHECKING
+import warnings
+
+from bigframes_vendored.pandas.core.computation.engines import ENGINES
+from bigframes_vendored.pandas.core.computation.expr import Expr, PARSERS
+from bigframes_vendored.pandas.core.computation.parsing import tokenize_string
+from bigframes_vendored.pandas.core.computation.scope import ensure_scope
+from bigframes_vendored.pandas.core.generic import NDFrame
+from bigframes_vendored.pandas.util._exceptions import find_stack_level
+from bigframes_vendored.pandas.util._validators import validate_bool_kwarg
+from pandas.core.dtypes.common import is_extension_array_dtype
+from pandas.io.formats.printing import pprint_thing
+
+if TYPE_CHECKING:
+    from pandas.core.computation.ops import BinOp
+
+
+def _check_engine(engine: str | None) -> str:
+    """
+    Make sure a valid engine is passed.
+
+    Parameters
+    ----------
+    engine : str
+        String to validate.
+
+    Raises
+    ------
+    KeyError
+      * If an invalid engine is passed.
+    ImportError
+      * If numexpr was requested but doesn't exist.
+
+    Returns
+    -------
+    str
+        Engine name.
+    """
+    from pandas.core.computation.check import NUMEXPR_INSTALLED
```

Comment:
Ditto, I doubt we actually want to use numexpr, right?

Reply:
removed